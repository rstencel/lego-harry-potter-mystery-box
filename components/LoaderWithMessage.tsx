import { FormattedMessage } from "react-intl"

interface LoaderWitMessageProps {
  messageId: string
  isVisible: boolean
  textColor?: string
}

const LoaderWithMessage = ({ messageId, isVisible, textColor = "white" }: LoaderWitMessageProps) => (
  <div style={{ visibility: isVisible ? "visible" : "hidden" }} className="loader-with-message">
    <div className="loader" />
    <p style={{ color: textColor, marginLeft: "1rem" }}>
      <FormattedMessage id={messageId} />
    </p>
  </div>
)

export default LoaderWithMessage