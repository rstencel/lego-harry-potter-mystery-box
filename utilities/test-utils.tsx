import React, { FC, ReactElement } from "react"
import { render, RenderOptions } from "@testing-library/react"
import { IntlProvider } from "react-intl"
import messages from "../translations/en"

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <IntlProvider locale="en" defaultLocale="en" messages={messages}>
    {children}
  </IntlProvider>
)

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options })

export { customRender as render }