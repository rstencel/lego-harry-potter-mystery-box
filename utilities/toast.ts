import { toast, TypeOptions } from 'react-toastify'

export const showToast = (message: string, type: TypeOptions, onCloseCallback?: () => void) => {
  toast.onChange(payload => {
    if (payload.status === "removed" && onCloseCallback) {
      onCloseCallback()
    }
  })
  
  toast(message, {
    type,
    position: "top-center",
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: false,
    pauseOnHover: false,
    draggable: false,
    progress: undefined,
    theme: "light"
  })
}