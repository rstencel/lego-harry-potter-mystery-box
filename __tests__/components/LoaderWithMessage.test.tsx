import LoaderWithMessage from "../../components/LoaderWithMessage"
import { render } from "../../utilities/test-utils"

describe("LoaderWithMessage", () => {
  describe("Should match snapshot", () => {
    it("- invisible", () => {
      const { container } = render(<LoaderWithMessage isVisible={false} messageId="home.draw.loading" />)
      expect(container).toMatchSnapshot()
    })
    
    it("- visible", () => {
      const { container } = render(<LoaderWithMessage isVisible messageId="home.draw.loading" />)
      expect(container).toMatchSnapshot()
    })
  })

  it("Should display message", () => {
    const { container } = render(<LoaderWithMessage isVisible messageId="home.draw.loading" />)
    expect(container).toHaveTextContent("Drawing minifigs ...")
  })
})
