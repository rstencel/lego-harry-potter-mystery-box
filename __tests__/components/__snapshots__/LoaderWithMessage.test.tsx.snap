// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderWithMessage Should match snapshot - invisible 1`] = `
<div>
  <div
    class="loader-with-message"
    style="visibility: hidden;"
  >
    <div
      class="loader"
    />
    <p
      style="color: white; margin-left: 1rem;"
    >
      Drawing minifigs ...
    </p>
  </div>
</div>
`;

exports[`LoaderWithMessage Should match snapshot - visible 1`] = `
<div>
  <div
    class="loader-with-message"
    style="visibility: visible;"
  >
    <div
      class="loader"
    />
    <p
      style="color: white; margin-left: 1rem;"
    >
      Drawing minifigs ...
    </p>
  </div>
</div>
`;
